src/wp-admin/includes/upgrade.php

```diff
@@ -2349,6 +2353,36 @@ function upgrade_640() {
 	}
 }
 
+/**
+ * Executes changes made in WordPress 6.5.0.
+ *
+ * @ignore
+ * @since 6.5.0
+ *
+ * @global int  $wp_current_db_version The old (current) database version.
+ * @global wpdb $wpdb                  WordPress database abstraction object.
+ */
+function upgrade_650() {
+	global $wp_current_db_version, $wpdb;
+
+	if ( $wp_current_db_version < 57200 ) {
+		$stylesheet = get_stylesheet();
+
+		// Set autoload=no for the previous all themes.
+		$theme_mods_options = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT option_name FROM $wpdb->options WHERE autoload = 'yes' AND option_name != %s AND option_name LIKE %s",
+				"theme_mods_$stylesheet",
+				$wpdb->esc_like( 'theme_mods_' ) . '%'
+			)
+		);
+
+		$autoload = array_fill_keys( $theme_mods_options, 'no' );
+		$autoload[ "theme_mods_$stylesheet" ] = 'yes';
```

Comment:
See above. If the `template` differs from `$stylesheet`, we should also set that to `yes` here.

More importantly though, why do we even need this line? The purpose of this function is to set old theme mods to `no`. We don't need to explicitly set the current theme mods to `yes` here, as that will already be the case.

Reply:
Addressed in https://github.com/WordPress/wordpress-develop/pull/5711/commits/8e7861d281b559ccbbbf65668e541af89afdb1c4